### PR TITLE
apps: added the possibility to enable falco in sc

### LINKED ISSUE #949: [3] Add falco in service cluster
**Is your feature request related to a problem? Please describe.**
We currently do not run falco in the service cluster. We should add it there as well to increase our security.

**Describe the solution you'd like**
Add falco in the service cluster and update the rules with relevant exceptions for the other pods we run in the service cluster (so that we do not get spammed with unnecessary alerts). Since we don't really know how many more rules we would need to edit or how much time that would take we want this task to be time boxed to 3 days. If there are more rules that needs updates then those should be noted down in a new issue. Try to focus on the rules that are causing the most amount of alerts.

**Describe alternatives you've considered**

**Additional context**

**Definition of done:**
- Falco is running in the service cluster
- We get alerts from falco in the service cluster
- The falco rules are updated with relevant exceptions (preferably upstream as well).
- An issue is created with any rule updates that was not completed in the time for this task.

### LINKED ISSUE #1100: falco - host alert
**Describe the bug**
Falco is generating the below host alert on a test cluster setup:

> {"output":": Notice Unexpected setuid call by non-sudo, non-root program (user=<NA> user_loginuid=-1 cur_uid= parent=<NA> command=<NA> uid=<NA> container_id=host image=<NA>) k8s.ns=<NA> k8s.pod=<NA> container=host k8s.ns=<NA> k8s.pod=<NA> container=host k8s.ns=<NA> k8s.pod=<NA> container=host k8s.ns=<NA> k8s.pod=<NA> container=host","priority":"Notice","rule":"Non sudo setuid","source":"syscall","tags":["mitre_privilege_escalation","users"],"time":"", "output_fields": {"container.id":"host","container.image.repository":null,"evt.arg.uid":"<NA>","evt.time":,"k8s.ns.name":null,"k8s.pod.name":null,"proc.cmdline":"<NA>","proc.pname":null,"user.loginuid":-1,"user.name":null,"user.uid":}}

**Expected behaviour**
Alert not being triggered or an exception added if it's a know behaviour that doesn't affect our security.

**Definition of Done**
- wait until we complete the [falco v0.33 upgrade task ](https://github.com/elastisys/compliantkubernetes-apps/issues/1017)
- check if the alert is still present (check after 24h or 48h)
- if still there investigate the issue and propose a solution

### PR DESCRIPTION
**Pull request by crssnd**
**What this PR does / why we need it**: to be able to install falco in sc

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #949 

**Special notes for reviewer**:
- I create another issue for the alert that I didn't manage to identify and fix 
    - #1100 
    - #1105
    - #1114 
    - #1115
    - #1116
    - #1119
    - #1120
- except the above alert, no other alerts are generated using a default test cluster

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [x] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [x] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

### CODE REVIEW

**Review by Xartos — APPROVED** [review]
LGTM

File: scripts/clean-sc.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ here="$(dirname "$(readlink -f "$0")")"
 "${here}/.././bin/ck8s" ops helmfile sc -l app!=cert-manager destroy
 
 # Clean up namespaces and any other resources left behind by the apps
-"${here}/.././bin/ck8s" ops kubectl sc delete ns dex opensearch-system harbor fluentd thanos influxdb-prometheus ingress-nginx monitoring velero kured
```
Comment (davidumea) [inline scripts/clean-sc.sh:20]: shouldn't velero still be "cleaned up" here?

Comment (crssnd) [reply scripts/clean-sc.sh:20]: I noticed that we delete the namespace again lower in the script, line 32

**Review by davidumea — APPROVED** [review]

**Review by robinAwallace — APPROVED** [review]

**Review by viktor-f — COMMENTED** [review]
Since you linked to https://github.com/elastisys/compliantkubernetes-apps/issues/1080:
I saw some more things that one should probably look at when I checked falco in wc in a running cluster. Is this something that you can also look at, or create new issue for it?

- Alert when postgres is spawning wal-g:
```
{"output":"09:49:46.578763685: Notice Database-related program spawned process other than itself (user=<NA> user_loginuid=-1 program=sh -c envdir \"/run/etc/wal-e.d/env\" wal-g wal-push \"pg_wal/0000002E000006300000002A\" parent=postgres container_id=086c88f1ef91 image=registry.opensource.zalan.do/acid/spilo-14) k8s.ns=postgres-system k8s.pod=long-running-13-a-0 container=086c88f1ef91 k8s.ns=postgres-system k8s.pod=long-running-13-a-0 container=086c88f1ef91 k8s.ns=postgres-system k8s.pod=long-running-13-a-0 container=086c88f1ef91 k8s.ns=postgres-system k8s.pod=long-running-13-a-0 container=086c88f1ef91","priority":"Notice","rule":"DB program spawned process","source":"syscall","tags":["database","mitre_execution","process"],"time":"2022-08-15T09:49:46.578763685Z", "output_fields": {"container.id":"086c88f1ef91","container.image.repository":"registry.opensource.zalan.do/acid/spilo-14","evt.time":1660556986578763685,"k8s.ns.name":"postgres-system","k8s.pod.name":"long-running-13-a-0","proc.cmdline":"sh -c envdir \"/run/etc/wal-e.d/env\" wal-g wal-push \"pg_wal/0000002E000006300000002A\"","proc.pname":"postgres","user.loginuid":-1,"user.name":null}}
```
- Alert about namespace change:
```
{"output":"09:51:41.088557396: Notice Namespace change (setns) by unexpected program (user=root user_loginuid=-1 command=nsenter -m/proc/1/ns/mnt -- test -f /var/run/reboot-required parent=kured k8s.ns=kured k8s.pod=kured-lj9qq container=8855c699d6d5 container_id=8855c699d6d5 image=docker.io/weaveworks/kured:1.9.1) k8s.ns=kured k8s.pod=kured-lj9qq container=8855c699d6d5","priority":"Notice","rule":"Change thread namespace","source":"syscall","tags":["mitre_lateral_movement","mitre_privilege_escalation","process"],"time":"2022-08-15T09:51:41.088557396Z", "output_fields": {"container.id":"8855c699d6d5","container.image.repository":"docker.io/weaveworks/kured","container.image.tag":"1.9.1","evt.time":1660557101088557396,"k8s.ns.name":"kured","k8s.pod.name":"kured-lj9qq","proc.cmdline":"nsenter -m/proc/1/ns/mnt -- test -f /var/run/reboot-required","proc.pname":"kured","user.loginuid":-1,"user.name":"root"}}
```

File: config/config/sc-config.yaml [hunk 2]
```diff
@@ -90,6 +90,13 @@ user:
         limits:
           cpu: 10m
           memory: 100Mi
+falco:
+  customRules:
+    grafana.yaml: |-
+      - macro: user_known_contact_k8s_api_server_activities
+        append: true
+        condition: or (container.image.repository in (quay.io/kiwigrid/k8s-sidecar, ghcr.io/dexidp/dex))
```
[the remark this answers is not in the document]
Comment (crssnd) [reply config/config/sc-config.yaml:98]: fixed

**Review by viktor-f — APPROVED** [review]

**Pull request merged**